engine/consensus/provider/engine_test.go

```diff
@@ -53,25 +53,38 @@ func (suite *Suite) SetupTest() {
 	localID := suite.identities[0].NodeID
 
 	suite.me.On("NodeID").Return(localID)
-	suite.state.On("Final").Return(suite.final)
+	suite.state.On("AtBlockID", mock.Anything).Return(suite.final).Maybe()
 	suite.final.On("Identities", mock.Anything).Return(
 		func(f flow.IdentityFilter) flow.IdentityList { return suite.identities.Filter(f) },
 		func(flow.IdentityFilter) error { return nil },
-	)
+	).Maybe()
 }
 
-// proposals submitted by remote nodes should not be accepted.
-func (suite *Suite) TestOnBlockProposal_RemoteOrigin() {
+// TestBroadcastProposal_Success should broadcast a valid block.
+func (suite *Suite) TestBroadcastProposal_Success() {
 
 	proposal := unittest.ProposalFixture()
-	// message submitted by remote node
-	err := suite.engine.onBlockProposal(suite.identities[1].NodeID, proposal)
-	suite.Assert().Error(err)
+	proposal.Header.ProposerID = suite.me.NodeID()
+
+	params := []interface{}{proposal}
+	for _, identity := range suite.identities {
+		// skip consensus nodes
+		if identity.Role == flow.RoleConsensus {
+			continue
+		}
+		params = append(params, identity.NodeID)
+	}
+
+	suite.conduit.On("Publish", params...).Return(nil).Once()
+
+	err := suite.engine.broadcastProposal(proposal)
+	suite.Assert().NoError(err)
 }
 
-func (suite *Suite) OnBlockProposal_Success() {
+func (suite *Suite) TestBroadcastProposal_Invalid() {
 
 	proposal := unittest.ProposalFixture()
+	proposal.Header.ProposerID = unittest.IdentifierFixture() // set a random proposer
```

Comment:
It's already random from `unittest.ProposalFixture` isn't it?

Reply:
It is - this is only for test readability